### PR TITLE
Project rename including methods, variables, etc

### PR DESCRIPTION
**Pull request by jayphelps**
@stefanpenner I could merge myself, but I think this needs another set of eyes as well as confirm the name change still feels right.

Besides the word *session* being replaced with *state*, I also renamed `endSessionFor()` -> `teardownStateFor()` because "endState" didn't seem to make sense and teardown is the opposite of setup.

image

### CODE REVIEW

**Comment by thoov** [comment]
@jayphelps @stefanpenner I do like sessions being changed to state but just a quick glance over this PR the app/states/* seemed kinda weird to me (could just be me). Also have we thought of something other than state-services as a project name? Possibly more generic like ember-state?

**Comment by jayphelps** [comment]
@thoov I think I know what your feeling. I had the same feeling with `routes/`. Ultimately, I think the point was primarily to get away from the word session asap so there isn't massive confusion with the already existing usage. [1](https://github.com/simplabs/ember-simple-auth), [2](https://github.com/Vestorly/torii), etc.

**Comment by stefanpenner** [comment]
@jayphelps can u rebase. I'll add you to npm and you can also release. Which npm username is yours?

File: tests/dummy/app/components/edit-email.js [hunk 1]
```diff
@@ -2,18 +2,18 @@ import Ember from 'ember';
 
 export default Ember.Component.extend({
   tagName: 'form',
-  session: Ember.computed('email', function() {
-    return this.editEmailService.sessionFor(this.get('email'));
+  state: Ember.computed('email', function() {
```
Comment (thoov) [inline tests/dummy/app/components/edit-email.js]: Here we should explicitly call:

```js
editEmailService: Ember.inject.service('edit-email')
```

As I believe this gives better context to how editEmailService is being set

Comment (thoov) [inline tests/dummy/app/components/edit-email.js]: This would also mean we can get rid of the initializer

**Comment by jayphelps** [comment]
@stefanpenner Rebased. I believe this is good to go now.

File: tests/dummy/app/components/edit-email.js (hunk 1 above)
Comment (jayphelps) [inline tests/dummy/app/components/edit-email.js]: :+1:

**Comment by jayphelps** [comment]
@stefanpenner npm username is [jayphelps](https://www.npmjs.com/~jayphelps)

**Comment by thoov** [comment]
:clap:

**Pull request merged**